core/server/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -49,18 +46,6 @@
   /** The unique id of each lock. */
   private static final AtomicLong LOCK_ID_GEN = new AtomicLong(0);
 
-  /** A pool of read write locks. */
-  private final ResourcePool<ClientRWLock> mLockPool = new ResourcePool<ClientRWLock>(
-      WorkerContext.getConf().getInt(Constants.WORKER_TIERED_STORE_BLOCK_LOCKS)) {
-    @Override
-    public void close() {}
-
-    @Override
-    protected ClientRWLock createNewResource() {
-      return new ClientRWLock();
-    }
-  };
```

Comment:
My understanding is that the concurrency is limited by the TThreadPoolServer's threadpool size (<=2048). This pool is only useful to limit the async lock usage. But I think we should control that based on the async persist rate limiter (or something else).